src/main/java/nixy/ui/Ui.java

```diff
@@ -0,0 +1,124 @@
+package nixy.ui;
+
+import java.util.Iterator;
+import java.util.Scanner;
+
+import nixy.exceptions.NixyException;
+import nixy.task.Task;
+import nixy.task.TaskList;
+
+/**
+ * Ui class is responsible for handling user interface.
+ * It provides methods to read input from the user and display messages to the user.
+ */
+public class Ui {
+
+    static final String HORIZONTAL_LINE = "____________________________________________________________";
+
+    /**
+     * Read input from the user.
+     */
+    public static String readInput() {
+        Scanner in = new Scanner(System.in);
+        String input = in.nextLine();
+        return input;
+    }
+
+    public static void showWelcome() {
+        PrintUtility.wrapPrintWithHorizontalLines("Hello! I'm Nixy", "What can I do for you?");
+    }
+
+    public static void showGoodbye() {
+        PrintUtility.wrapPrintWithHorizontalLines("Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Display the list of tasks to the user.
+     * @param tasks The list of tasks to display.
+     */
+    public static void showList(TaskList tasks) {
+        utilDisplayList(tasks, "Here are the tasks in your list:");
+    }
+
+    /**
+     * Displays the list of tasks that find retrieved.
+     *
+     * @param tasks The list of found tasks.
+     */
+    public static void showMatchingList(TaskList tasks) {
+        utilDisplayList(tasks, "Here are the matching tasks in your list:");
+    }
+
+    /**
+     * Helper method to display a list of tasks with a message.
+     * The list is displayed with a number prefix.
+     *
+     * @param tasks The list of tasks to display.
+     * @param message The message to display before the list.
+     */
+    private static void utilDisplayList(TaskList tasks, String message) {
```

Comment:
Maybe this could be `displayList` instead? Methods typically start with verbs that describes the action that the method performs